pkg/cmd/pulumi/new_test.go

```diff
@@ -818,3 +818,49 @@ func TestErrorIfNotEmptyDirectory(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateStackRefAndProjectName(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		projectName string
+		stackRef    string
+		valid       bool
+	}{
+		{
+			projectName: "foo",
+			stackRef:    "foo",
+			valid:       true,
+		},
+		{
+			projectName: "fooo",
+			stackRef:    "org/foo/dev",
+			valid:       false,
+		},
+		{
+			projectName: "",
+			stackRef:    "org/foo/dev",
+			valid:       true,
+		},
+		{
+			projectName: "foo",
+			stackRef:    "",
+			valid:       true,
+		},
```

Comment:
optional: can omit zero-value fields.
```suggestion
		},
		{
			stackRef:    "org/foo/dev",
			valid:       true,
		},
		{
			projectName: "foo",
			valid:       true,
		},
```

Reply:
We could, but I feel that it would be nicer to show this implicit detail.